### PR TITLE
Add built-in GDScript keywords to the class reference

### LINKED ISSUE #2361: Searching help for keywords / operators / built-in types should bring up a page about the term as the first result
### Describe the project you are working on
3D puzzle game

### Describe the problem or limitation you are having in your project
Searching the help for keywords / operators / built-in types / etc that aren't classes doesn't return the help page you expect.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
The feature I would like to see is that all language keywords are given an entry in both the online help and the in-editor help system such that if I search for a keyword (for, while, range etc) a page explaining that term comes up as the first search result. It's not logical (or the easiest to remember) that the usage of these terms is found in the GDScript -> GDScript basics page.
Having the help return a page as the first response for the term you enter would make Godot a lot easier to use, particularly for beginners.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Press F1, search for "for" and a page explaining how to use the "for" command is returned as the **first** search result.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No

### Is there a reason why this should be core and not an add-on in the asset library?
Getting relevant help is critical for efficient use of the engine.

### PR DESCRIPTION
**Pull request by Calinou**
These are part of `@GDScript` and appear in search results as part of a new `builtin` documentable type. (`keyword` is not used, as `keywords` is already the name of an optional attribute for most documentable types.)

Keywords can now be <kbd>Ctrl</kbd> + clicked in the script editor to go to their class reference description.

- This closes https://github.com/godotengine/godot-proposals/issues/2361.

## Preview

https://github.com/user-attachments/assets/5caf0619-0fea-44af-819c-6f149b32f576

## TODO

- [ ] Fill in descriptions.
- [ ] Define parameters for keywords when relevant.
- [ ] Update `doc_status.py` to allow tracking progress on builtins.
- [ ] Test `make_rst.py`'s reStructuredText generation on the docs website.

Filling in the descriptions could be done in a separate PR if we want, like we did for https://github.com/godotengine/godot/pull/48548 (where https://github.com/godotengine/godot/pull/63870 added the descriptions).

In a future PR, we can look into using a similar approach to [document shader keywords in the class reference](https://github.com/godotengine/godot-proposals/issues/3767), in addition to built-ins (which would probably be considered member variables there).

### CODE REVIEW

File: modules/gdscript/gdscript_editor.cpp [hunk 1]
```diff
@@ -434,6 +434,63 @@ void GDScriptLanguage::get_recognized_extensions(List<String> *p_extensions) con
 	p_extensions->push_back("gd");
 }
 
+void GDScriptLanguage::get_public_keywords(List<MethodInfo> *p_keywords) const {
+	Vector<String> keywords = {
+		// Control flow.
```
Comment (dalexeev) [inline modules/gdscript/gdscript_editor.cpp:439]: It would be nice to add a comment with a link to `GDScriptLanguage::get_reserved_words()` and a note that function- and constant-like keywords are excluded.

https://github.com/godotengine/godot/blob/80db886ec251702aacec99d03628f6a0b8c44297/modules/gdscript/gdscript.cpp#L2719-L2721

File: modules/gdscript/doc_classes/@GDScript.xml [hunk 2]
```diff
@@ -280,6 +280,205 @@
...
+			<return type="void" />
+			<description>
+			</description>
+		</builtin>
+		<builtin name="yield">
+			<return type="void" />
+			<description>
+				Reserved for future use. This keyword causes a script error when used.
+			</description>
+		</builtin>
```
Comment (HolonProduction) [inline modules/gdscript/doc_classes/@GDScript.xml:480]: Is `yield` really reserved for future use? I thought it was reserved to show a proper warning for legacy code, so more like reserved for compatibility.

Comment (Mickeon) [reply modules/gdscript/doc_classes/@GDScript.xml:480]: A bit of both, but it's mostly reserved for legacy reasons. There are no plans to bring coroutines with `yield` back in the foreseeable future, but there is a desire for them. Indeed the current description is a bit misleading.

Comment (HolonProduction) [reply modules/gdscript/doc_classes/@GDScript.xml:480]: That's a bit of my problem with showing those reserved words in general, it might be easy to misunderstand them as "this feature is confirmed and will be implemented soon".

**Comment by Mickeon** [comment]
I know it's the easiest way to accomplish this, but something about it... It doesn't feel right for `builtins` to be something that can be documented for every single class, even though they're GDScript-exclusive concepts. 

Furthermore, because only keywords are documented, the class reference still lacks a general point of reference for literals or similar. This may be fine, and for the most part we can direct users to the [GDScript reference](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html), but it still means that adding just keywords doesn't accomplish the goal of an all-encompassing guide built right inside the editor.

**Comment by HolonProduction** [comment]
So I just did a small bit of coding using this, and I'm not a fan of the Ctrl+Click lookup on keywords. In my opinion it adds too much noise when trying to lookup other stuff. Since keywords (especially `var`) might be on the screen in high amounts (10+) just moving the cursor to something I want to lookup will cause a lot of flickering underlines all over the screen.

It might be helpful for beginners, but knowing GDScript it feels more like a downgrade of the script editor to me.

Maybe we could prevent the underline highlighting for certain lookup targets :thinking:

**Comment by aXu-AP** [comment]
I remember when starting writing gdscript I had to check a good few times what was the syntax for `for` and `match` before getting it memorized. Under context menu it could go unnoticed, in that case it might as well not exist.

**Comment by Mickeon** [comment]
> So I just did a small bit of coding using this, and I'm not a fan of the Ctrl+Click lookup on keywords. In my opinion it adds too much noise when trying to lookup other stuff. 

I think the "noise" can be severely mitigated and the existing behaviour should no longer be an issue. When hovering over keywords, **not all** occurrences of that keyword should be highlighted.
Keywords are not class members, variables, or similar. For those, in layman's terms, the same name means the same reference.

File: core/doc_data.h [hunk 3]
```diff
@@ -939,6 +948,14 @@ class DocData {
 				dict["properties"] = properties;
 			}
 
+			if (!p_doc.builtins.is_empty()) {
+				Array builtins;
+				for (int i = 0; i < p_doc.builtins.size(); i++) {
+					builtins.push_back(MethodDoc::to_dict(p_doc.builtins[i]));
```
Comment (AThousandShips) [inline core/doc_data.h]:
```suggestion
				for (const MethodDoc &builtin : p_doc.builtins) {
					builtins.push_back(MethodDoc::to_dict(builtin));
```
